warehouse/query-core/src/main/java/datawave/query/jexl/visitors/JexlFormattedStringBuildingVisitor.java

```diff
@@ -310,21 +708,139 @@ public Object visit(ASTReferenceExpression node, Object data) {
+        StringBuilder sb = (StringBuilder) data;
+        for (int i = 0; i < node.jjtGetNumChildren(); i++) {
+            node.jjtGetChild(i).jjtAccept(this, sb);
+        }
+        
+        return sb;
+    }
+    
+    @Override
+    public Object visit(ASTSizeMethod node, Object data) {
+        StringBuilder sb = (StringBuilder) data;
+        
+        decorator.apply(sb, node);
+        
+        node.childrenAccept(this, sb);
+        return sb;
+    }
+    
+    @Override
+    public Object visit(ASTMulNode node, Object data) {
+        StringBuilder sb = (StringBuilder) data;
+        int numChildren = node.jjtGetNumChildren();
+        for (int i = 0; i < numChildren; i++) {
+            node.jjtGetChild(i).jjtAccept(this, sb);
+            if (i < numChildren - 1) {
+                decorator.apply(sb, node);
+            }
+        }
+        
+        return sb;
+    }
+    
+    @Override
+    public Object visit(ASTDivNode node, Object data) {
+        StringBuilder sb = (StringBuilder) data;
+        int numChildren = node.jjtGetNumChildren();
+        for (int i = 0; i < numChildren; i++) {
+            node.jjtGetChild(i).jjtAccept(this, sb);
+            if (i < numChildren - 1) {
+                decorator.apply(sb, node);
+            }
+        }
+        
+        return sb;
+    }
+    
+    @Override
+    public Object visit(ASTModNode node, Object data) {
+        StringBuilder sb = (StringBuilder) data;
+        int numChildren = node.jjtGetNumChildren();
+        for (int i = 0; i < numChildren; i++) {
+            node.jjtGetChild(i).jjtAccept(this, sb);
+            if (i < numChildren - 1) {
+                decorator.apply(sb, node);
+            }
+        }
+        
+        return sb;
+    }
+    
+    @Override
+    public Object visit(ASTAssignment node, Object data) {
+        boolean requiresParens = !(node.jjtGetParent() instanceof ASTReferenceExpression);
+        StringBuilder sb = (StringBuilder) data;
+        if (requiresParens)
+            sb.append('(');
+        for (int i = 0; i < node.jjtGetNumChildren(); ++i) {
+            node.jjtGetChild(i).jjtAccept(this, sb);
+            decorator.apply(sb, node, i);
+        }
+        if (requiresParens) {
+            sb.append(')');
+        }
+        return sb;
+    }
+    
+    @Override
+    public Object visit(ASTUnaryMinusNode node, Object data) {
+        StringBuilder sb = (StringBuilder) data;
+        
+        decorator.apply(sb, node);
+        
+        node.childrenAccept(this, sb);
+        return sb;
+    }
+    
+    /**
+     * Returns a new list of QueryMetrics which is identical to the given list except the query and query plan for each metric are formatted according to
+     * {@link #buildDecoratedQuery(JexlNode, boolean, JexlQueryDecorator) buildDecoratedQuery} with an {@link HtmlDecorator}
+     * 
+     * @param metrics
+     */
+    public static void formatMetrics(List<? extends BaseQueryMetric> metrics) {
```

Comment:
This seems like a weird home for this method.  Should this not just be a method in the ShardTableQueryMetricHandler?  The StringBuilder shouldn't have to concern itself with query metrics.

Reply:
Yes I agree. It was meant to just be a temporary spot but I never got around to changing it. Thanks for pointing this out. I'll move it to ShardTableQueryMetricHandler.